pkg/operators/oci/oci.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocioperator
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/oci"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	validateMetadataParam = "validate-metadata"
+	authfileParam         = "authfile"
+	insecureParam         = "insecure"
+	pullParam             = "pull"
+	pullSecret            = "pull-secret"
+)
+
+// OciHandler bridges our legacy operator system with the image based gadgets
+// once we remove the legacy gadgets, this operator should be called directly as if it
+// were the gadget
+type OciHandler struct{}
+
+func (o *OciHandler) ParamDescs() params.ParamDescs {
+	return params.ParamDescs{
+		// Hardcoded for now
+		{
+			Key:          authfileParam,
+			Title:        "Auth file",
+			DefaultValue: oci.DefaultAuthFile,
+			TypeHint:     params.TypeString,
+		},
+		{
+			Key:          validateMetadataParam,
+			Title:        "Validate metadata",
+			Description:  "Validate the gadget metadata before running the gadget",
+			DefaultValue: "true",
+			TypeHint:     params.TypeBool,
+		},
+		{
+			Key:          insecureParam,
+			Title:        "Insecure connection",
+			Description:  "Allow connections to HTTP only registries",
+			DefaultValue: "false",
+			TypeHint:     params.TypeBool,
+		},
+		{
+			Key:          pullParam,
+			Title:        "Pull policy",
+			Description:  "Specify when the gadget image should be pulled",
+			DefaultValue: oci.PullImageMissing,
+			PossibleValues: []string{
+				oci.PullImageAlways,
+				oci.PullImageMissing,
+				oci.PullImageNever,
+			},
+			TypeHint: params.TypeString,
+		},
+		{
+			Key:         pullSecret,
+			Title:       "Pull secret",
+			Description: "Secret to use when pulling the gadget image",
+			TypeHint:    params.TypeString,
+		},
+	}
+}
```

Comment:
I'm wondering how this will be handled. We can't rely on `GetOCIGadgetInfo` because these parameters are needed to pull the image, so I suppose somehow we'll have to hardcode them in the client so they're already available when calling `GetOCIGadgetInfo`. 

Also, the commit message introducing this file says it's not an operator anymore, so how will be these parameters added to the CLI? 

I gave it a try on https://github.com/inspektor-gadget/inspektor-gadget/commit/4f754cfc2cde7da677caff913cb12dc3217125d2, is this aligned with the ideas you had?